src/main/scala/chisel3/simulator/HierarchicalValue.scala

```diff
@@ -0,0 +1,54 @@
+package chisel3
+package simulator
+
+import chisel3.util.log2Ceil
+import chisel3.experimental.SourceInfo
+
+import svsim.Simulation
+
+trait HierarchicalValue {
+  val gen: Data
```

Comment:
```suggestion
  def gen: Data
```
Generally, it's best to make any virtual methods a `def` and let implementers decide what to do